### DISCUSSION TITLE
Error: Signing and encryption must use the same type of key

### DISCUSSION THREAD

**Discussion by lexxvir** [mcu-tools/mcuboot#1353] [Q&A] [answered]
Hi there,

Thank you for the excellent tool!

If found that imgtool.py restricts usage of different type of keys for signing & encryption (see the error from the subject).
Could you please suggest why is such a restriction exist?

In my case I'd like to use RSA2048 for signing & ecdsa-p256 for encryption.

**Comment by d3zd3z** [comment, marked as answer]
It might be possible to do this, but my attempts to get a working configuration for Mbed TLS to allow both of these was not successful.
